sbi/inference/potentials/posterior_based_potential.py

```diff
@@ -99,14 +100,23 @@ def __call__(self, theta: Tensor, track_gradients: bool = True) -> Tensor:
             )
 
         theta = ensure_theta_batched(torch.as_tensor(theta))
+
         theta, x = theta.to(self.device), self.x_o.to(self.device)
```

Comment:
x_o.to(self.device) is redundant, the `BasePotential` will always keep `x_o` on the device.